core/eolearn/core/eodata_io.py

```diff
@@ -155,6 +156,136 @@ def walk_eopatch(eopatch, patch_location, features=...):
+    if timeless_op not in allowed_timeless_op:
+        raise ValueError('timeless_op "%s" is invalid, must be one of %s' % (timeless_op, allowed_timeless_op))
+    if time_dependent_op not in allowed_time_dependent_op:
+        raise ValueError('time_dependent_op "%s" is invalid, must be one of %s'
+                         % (time_dependent_op, allowed_time_dependent_op))
+
+
+def _perform_concat_operation(eopatch, eopatch_content_concat, sorted_indices):
+    """ Performs the temporal concatenation of non-duplicate timestamps of time-dependent features.
+    """
+    for feat, arrays in eopatch_content_concat.items():
+        ftype, _ = feat
+        if ftype != FeatureType.VECTOR and sorted_indices:
+            eopatch[feat] = np.concatenate(arrays, axis=0)[sorted_indices]
+        else:
+            eopatch[feat] = pd.concat([array for array in arrays if not array.empty]).sort_values('TIMESTAMP')
+
+    return eopatch
+
+
+def _perform_merge_operation(eopatch, eopatch_content, sorted_indices, time_dependent_op, timeless_op):
+    """ Performs the merging of duplicate timestamps of time-dependent features and of timeless features.
+    """
+    for feat, arrays in eopatch_content.items():
+        ftype, _ = feat
+        if ftype.is_time_dependent() and ftype != FeatureType.VECTOR and sorted_indices:
+            if time_dependent_op == 'mean':
+                eopatch[feat] = np.nanmean(arrays, axis=0)[sorted_indices]
+            elif time_dependent_op == 'median':
+                eopatch[feat] = np.nanmedian(arrays, axis=0)[sorted_indices]
+            elif time_dependent_op == 'min':
+                eopatch[feat] = np.nanmin(arrays, axis=0)[sorted_indices]
+            elif time_dependent_op == 'max':
+                eopatch[feat] = np.nanmax(arrays, axis=0)[sorted_indices]
+        if ftype.is_timeless():
+            if timeless_op == 'mean':
+                eopatch[feat] = np.nanmean(arrays, axis=0)
+            if timeless_op == 'median':
+                eopatch[feat] = np.nanmedian(arrays, axis=0)
+            if timeless_op == 'min':
+                eopatch[feat] = np.nanmin(arrays, axis=0)
+            if timeless_op == 'max':
+                eopatch[feat] = np.nanmax(arrays, axis=0)
+            else:
+                eopatch[feat] = arrays[0]
+
+    return eopatch
+
+
+def _check_duplicate_timestamps(eopatch_content, duplicate_timestamp):
+    """ Checks that no duplicate timestamps with different values exist
+    """
+    for dup_time in duplicate_timestamp:
+        for dup_feat in eopatch_content.keys():
+            arrays = [eopatch_content[feat][eopatch_content[(FeatureType.TIMESTAMP, ...)].index(dup_time)]
+                      for eop_idx, (feat, _) in enumerate(eopatch_content.items())
+                      if dup_time in eopatch_content[(FeatureType.TIMESTAMP, ...)] and dup_feat == feat]
+            if any(not deep_eq(x, y) for i, x in enumerate(arrays) for j, y in enumerate(arrays) if i != j):
```

Comment:
These features checks should be done in the same place as merging operations are done (i.e. `_perform_merge_operation` in your case). After all, you can imagine "take 1 if all equal otherwise raise an error" as an operation, similar to min, max, mean, median.